src/mainwindow.cpp

```diff
@@ -6667,26 +6667,51 @@ void MainWindow::on_exploreView_customContextMenuRequested(QPoint pos)
     QMenu menu(ui->exploreView);
     QAction *action;
     /* Get path from index */
-    auto index   = ui->exploreView->selectionModel()->selectedIndexes()[0];
-    auto path    = getPathFromExplorerViewIndexModel(index);
-    bool is_file = !QDir(path).exists();
+    auto indexes   = ui->exploreView->selectionModel()->selectedIndexes();
+    auto index     = indexes[0];
+    auto path      = getPathFromExplorerViewIndexModel(index);
+    bool is_file   = !QDir(path).exists();
 
     if (is_file)
     {
-        action = new QAction("&Load", this);
-         connect(action, &QAction::triggered, this, [this](){
-             auto index = ui->exploreView->selectionModel()->selectedIndexes()[0];
-             on_exploreView_activated(index);
-         });
+        action = new QAction("&Load selected", this);
+        connect(action, &QAction::triggered, this, [this, indexes](){
+            qDebug() << __FUNCTION__;
+
+            QSet<QString> pathsSet;
+            auto selectedIndexes = indexes;
+
+            for (auto &index : selectedIndexes)
+            {
+               QString path = getPathFromExplorerViewIndexModel(index);
+
+               if (path.toLower().endsWith(".dlt"))
+               {
+                   pathsSet.insert(path);
```

Comment:
Loading multiple file is important to open them in the right order, so that they are sorted by time. QSet does not garantue the order. Please use another class e.g. QList where the order is guaranteed.

Reply:
Done.
Added possibility to change sorting type and order in explorer view, and the most important - replacing QSet with QList to maintain the right order.